doc/adr/Which-branch-holds-what-is-reflected-in-live.md

```diff
@@ -0,0 +1,112 @@
+# ADR - Which branch holds what is reflected in Live?
+
+Date: 4/12/2019
+
+## Status
+
+Accepted
+
+## Context
+
+DSO keeps both 'product' code and 'config' code in github. Currently this all tends to be stored together in one repo.
+
+Product code developers traditionally signal to the release process that a release is _ready for release_ by 'tagging' a commit and pushing the tag. Often the tag is created by the CI tooling itself so that a tagged commit can only ever be released if it has been through an assurance process and is therefore _ready for deployment_.
+
+Infrastructure teams working with 'configuration repos' tend to use branches or folders/files to represent which _released_ version of the infrastructure/ and/or application code is currently _deployed_ in which environment. This means that these live branches must be protected. The PR merge process is the means by which a decision to deploy is made.  
+
+## Decision
+
+We should manage the 'release process' of products and the 'deployment process' of services according to their own, differing requirements, using industry standard automation processes to protect service while increasing development velocity and ease of release & deploy. 
+
+### Product code
+
+#### Versioning/Packaging
+
+Tested, released product code such as a terraform project, a helm chart, or an npm package should be stored in either 
+
+1) The appropriate write-only repository type
+	- ie https://www.npmjs.com/ for nodejs packages
+	- docker hub or ACR for docker images
+	- etc
+
+or 
+
+2) In github as a specific tagged commit.
+	
+These two constructs are immutable (write-only), and so form a contract with their consumer that whatever testing has been performed against them has not been invalidated by hidden change. 
+
+Development of product code should be done directly on master, so that everyone is integrating continuously.
+
+#### References / Depencencies
+
+*Released* products MUST specify their dependencies using immutable references, or must vendor dependencies in so that the resulting deploys are safe.
```

Comment:
'must vendor dependencies' < sorry, did not understand what you're trying to say here.